spec/controllers/comments_controller_spec.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe CommentsController do
+  render_views
+
+  let(:user) { create(:user) }
+  let(:task) { create(:task) }
+
+  before { sign_in user }
+
+  describe 'index' do
+    it 'renders load_comments view' do
+      get :index, xhr: true, format: :js, params: {task_id: task.id}
```

Comment:
That method call is totally fine, and you are absolutely right about all arguments you pass here! Both, `format` and `xhr` are required, as the code won't work otherwise.

A minor suggestion here as well: I would suggest reordering the arguments. Sure, this doesn't have any technical implications but rather should assist in reading the code. My suggestion:

```suggestion
      get :index, params: {task_id: task.id}, format: :js, xhr: true
```

This way, we first complete the URL (using the `params`). Then, we specify the format and finally how the request should be performed (as XHR). This order also reflects how common those parameters are added: A parameter is very common, the format somewhat common, and the XHR rather not common.